### ISSUE TITLE
dcm anonymisation leads to invalid birthdate

### ISSUE THREAD

**Issue by bjeurissen** [MRtrix3/mrtrix3#1518]
```
dcminfo -all file.ima
```
:
```
...
[DCM] 0010 0020 LO       10     1304   PatientID                              QP-1556375 
[DCM] 0010 0021 LO        8     1322   IssuerOfPatientID                      ISSUER 
[DCM] 0010 0030 DA        8     1338   PatientBirthDate                       1981/03/02
...
```

```
dcmedit -anon file.ima
dcminfo -all file.ima
```
:
```
...
[DCM] 0010 0020 LO       10     1304   PatientID                              QP-1556375 
[DCM] 0010 0021 LO        8     1322   IssuerOfPatientID                      ISSUER 
dcminfo: [ERROR] error converting string "    "
[DCM] 0010 0030 DA        8     1338   PatientBirthDate                       invalid entry
...
```

**Comment by jdtournier** [comment]
Pretty sure that just means it's empty...? `dcmedit` just pads it out with binary zeros... I guess we could change the display from `invalid entry` to just `[no data]` if that's the issue...?

**Comment by jdtournier** [comment]
Although I agree the preceding error message is annoying. I'll take a look.

**Comment by jdtournier** [comment]
OK, that's basically an exception thrown from the `get_date()` call [at this point](https://github.com/MRtrix3/mrtrix3/blob/master/core/file/dicom/element.cpp#L414). Simplest fix is to relegate the [display of the exception message](https://github.com/MRtrix3/mrtrix3/blob/master/core/file/dicom/element.cpp#L434) to INFO or DEBUG level, i.e.:
```
434          e.display(3);
```
Which would at least silence the ERROR message. We'd still have `invalid entry` shown, is that a problem?

**Comment by bjeurissen** [comment]
I think the main issue is that a file that ``dcmedit`` creates, results in an ERROR on subsequent use. It is also unusual to have an (unfatal) ERROR that does not result in abortion of the command. My worry was mostly that it will confuse users and headless error checking that people might have in place.

**Comment by jdtournier** [comment]
So basically, the fix I'm suggesting above would sort out the issue, as far as you're concerned?

**Comment by bjeurissen** [comment]
Yes, my main issue was that our own anonymization leads to an error. Either it was not really an error and then we need to silence the error; or it was really an error and then we need to change the way we anonymize the date.

**Comment by Lestropie** [comment]
Closed by #1565.
